core/domain/collection_services.py

```diff
@@ -735,6 +743,22 @@ def publish_collection_and_update_user_profiles(committer_id, col_id):
             contributor, contribution_time_msec)
 
 
+def unpublish_collection_update_user_profiles(committer_id, col_id):
```

Comment:
I don't think we actually want this function. I don't think we should expose a general-purpose function for unpublishing collections to the rest of the code base (we don't for explorations, for instance). Unpublishing is generally a protected feature that regular users cannot use and it should be carefully protected. Only the collection_editor controller probably needs to be changed.